src/main/scala/viper/gobra/frontend/info/Info.scala

```diff
@@ -6,82 +6,268 @@
 
 package viper.gobra.frontend.info
 
+import com.typesafe.scalalogging.LazyLogging
 import org.bitbucket.inkytonik.kiama.relation.Tree
-import org.bitbucket.inkytonik.kiama.util.Source
-import viper.gobra.ast.frontend.{PNode, PPackage}
-import viper.gobra.frontend.Config
-import viper.gobra.frontend.PackageResolver.{AbstractImport, AbstractPackage}
+import org.bitbucket.inkytonik.kiama.util.Messaging.{Messages, message}
+import org.bitbucket.inkytonik.kiama.util.{Position, Source}
+import scalaz.EitherT
+import scalaz.Scalaz.futureInstance
+import viper.gobra.ast.frontend.{PImport, PNode, PPackage}
+import viper.gobra.frontend.{Config, Job, TaskManager}
+import viper.gobra.frontend.PackageResolver.{AbstractImport, AbstractPackage, BuiltInImport, BuiltInPackage, RegularImport}
+import viper.gobra.frontend.Parser.{ParseResult, ParseSuccessResult}
+import viper.gobra.frontend.TaskManagerMode.{Lazy, Parallel, Sequential}
 import viper.gobra.frontend.info.implementation.TypeInfoImpl
 import viper.gobra.frontend.info.implementation.typing.ghost.separation.{GhostLessPrinter, GoifyingPrinter}
-import viper.gobra.reporting.{CyclicImportError, TypeCheckDebugMessage, TypeCheckFailureMessage, TypeCheckSuccessMessage, TypeError, VerifierError}
+import viper.gobra.reporting.{CyclicImportError, ParserError, TypeCheckDebugMessage, TypeCheckFailureMessage, TypeCheckSuccessMessage, TypeError, VerifierError}
+import viper.gobra.util.{GobraExecutionContext, Violation}
 
-import scala.collection.immutable.ListMap
+import java.security.MessageDigest
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.{ConcurrentHashMap, ConcurrentMap}
+import scala.concurrent.Future
 
-object Info {
+object Info extends LazyLogging {
```

Comment:
I have seen that you extneded LazyLogging in some places. I only have a rough idea why. Please add a comment to the code explaining why this is necessary.